cameraview/src/main/java/com/otaliastudios/cameraview/engine/Camera2Engine.java

```diff
@@ -902,8 +902,44 @@ public void setPlaySounds(boolean playSounds) {
     }
 
     @Override
-    public void setZoom(float zoom, @Nullable PointF[] points, boolean notify) {
-        // TODO
+    public void setZoom(final float zoom, final @Nullable PointF[] points, final boolean notify) {
+        mHandler.run(new Runnable() {
+            @Override
+            public void run() {
+                if (!mCameraOptions.isZoomSupported()) return;
+
```

Comment:
Can you also check that getEngineState() == STATE_STARTED here? If not started let's return

Reply:
Also anytime you return from this function, you should call mZoomOp.end() (this is needed for tests)

Reply:
done